### PR TITLE
[FIX] *: Bad context keys in smart button

### PR DESCRIPTION
**Pull request by robinengels**
This reverts commit 60da4ee

Current behavior:
When adding a group_by:[opportunity_id] key to the context of the quotations page
the delivery smart button crash when there is more than 1 delivery

Steps to reproduce:
 -Add 'group_by':['opportunity_id'] in context of the quotations page
 -Refresh the page
 -Create a sale with more than 1 delivery
 -When you click on the delivery smart button the app crash

opw-2658775

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by robinengels** [comment]
@Julien00859

**Comment by Julien00859** [comment]
The resulting diff:

```diff
diff --git a/addons/sale_stock/models/sale_order.py b/addons/sale_stock/models/sale_order.py
index be522ddb3e2..8494d0dadce 100644
--- a/addons/sale_stock/models/sale_order.py
+++ b/addons/sale_stock/models/sale_order.py
@@ -204,7 +204,7 @@ class SaleOrder(models.Model):
             picking_id = picking_id[0]
         else:
             picking_id = pickings[0]
-        action['context'] = dict(self._context, default_partner_id=self.partner_id.id, default_picking_type_id=picking_id.picking_type_id.id, default_origin=self.name, default_group_id=picking_id.group_id.id)
+        action['context'] = dict(action.get('context', {}), default_partner_id=self.partner_id.id, default_picking_type_id=picking_id.picking_type_id.id, default_origin=self.name, default_group_id=picking_id.group_id.id, group_by=[])
         return action

     def action_cancel(self):
```

**Review by Julien00859 — CHANGES_REQUESTED** [review]
The context is still not evaluated

File: addons/sale_stock/models/sale_order.py [hunk 1]
```diff
@@ -204,7 +205,14 @@ def action_view_delivery(self):
             picking_id = picking_id[0]
         else:
             picking_id = pickings[0]
-        action['context'] = dict(action.get('context', {}), default_partner_id=self.partner_id.id, default_picking_type_id=picking_id.picking_type_id.id, default_origin=self.name, default_group_id=picking_id.group_id.id, group_by=[])
+        eval_ctx = dict(self.env.context)
+        print(eval_ctx)
```
Comment (Julien00859) [inline addons/sale_stock/models/sale_order.py:209]: oops

**Comment by robinengels** [comment]
@odoo/rd-security

**Review by Julien00859 — APPROVED** [review]

**Comment by odony** [comment]
If your commit says it's a revert, it should just revert the original one, not contain additional changes. Even more so if you broke stuff and you need to undo that quickly.

Also, the commit you want to revert is 0e6aa2c8c110c81b2a9e5b2afc1558e440154495, as 60da4eeec324f2097c52f00793222c067aef756b is not a merged commit. Maybe refer to the PR number instead: #79328

**Comment by odony** [comment]
sooo, this is becoming urgent because it impacts customer in production

**Comment by odony** [comment]
Sorry, original patch had to be quickly reverted at 5358ec76ce0cc99b128451188c958a21d6630d08. You could change this PR into an actual bugfix for the original problem, or close it and address that in a new PR.

**Comment by Julien00859** [comment]
Thank you and sorry Olivier. I had saved the commit elsewhere, re-pushing it.

**Pull request closed without merging**